api/v1/postgres_types.go

```diff
@@ -384,99 +389,60 @@ func (p *Postgres) ToPeripheralResourceNamespace() string {
-			Labels:    map[string]string{UIDLabelName: string(p.UID)},
-		},
-		Spec: ZalandoPostgresSpec{
-			MaintenanceWindows: func() []MaintenanceWindow {
-				if p.Spec.Maintenance == nil {
-					return nil
-				}
-				isEvery := p.Spec.Maintenance.Weekday == All
-				return []MaintenanceWindow{
-					{Everyday: isEvery,
-						Weekday: func() time.Weekday {
-							if isEvery {
-								return time.Weekday(0)
-							}
-							return time.Weekday(p.Spec.Maintenance.Weekday)
-						}(),
-						StartTime: p.Spec.Maintenance.TimeWindow.Start,
-						EndTime:   p.Spec.Maintenance.TimeWindow.End,
-					},
-				}
-			}(),
-			NumberOfInstances: p.Spec.NumberOfInstances,
-			PostgresqlParam:   PostgresqlParam{PgVersion: p.Spec.Version},
-			Resources: func() *Resources {
-				if p.Spec.Size.CPU == "" {
-					return nil
-				}
-				return &Resources{
-					ResourceRequests: &ResourceDescription{
-						CPU:    p.Spec.Size.CPU,
-						Memory: p.Spec.Size.Memory,
-					},
-					ResourceLimits: &ResourceDescription{
-						CPU:    p.Spec.Size.CPU,
-						Memory: p.Spec.Size.Memory,
-					},
-				}
-			}(),
-			TeamID: p.generateTeamID(),
-			Volume: Volume{Size: p.Spec.Size.StorageSize},
-		},
+func (p *Postgres) ToUnstructuredZalandoPostgresql(z *zalando.Postgresql) (*unstructured.Unstructured, error) {
+	if z == nil {
+		z = &zalando.Postgresql{}
 	}
-}
+	z.TypeMeta = ZalandoPostgresqlTypeMeta
+	z.Namespace = p.ToPeripheralResourceNamespace()
+	z.Name = p.ToPeripheralResourceName()
+	z.Labels = p.ToZalandoPostgresqlMatchingLabels()
+
+	z.Spec.NumberOfInstances = p.Spec.NumberOfInstances
+	z.Spec.PostgresqlParam.PgVersion = p.Spec.Version
+	z.Spec.Resources.ResourceRequests.CPU = p.Spec.Size.CPU
+	z.Spec.Resources.ResourceRequests.Memory = p.Spec.Size.Memory
+	z.Spec.Resources.ResourceLimits.CPU = p.Spec.Size.CPU
+	z.Spec.Resources.ResourceLimits.Memory = p.Spec.Size.Memory
+	z.Spec.TeamID = p.generateTeamID()
+	z.Spec.Volume.Size = p.Spec.Size.StorageSize
 
-func (p *Postgres) ToPatchedZalandoPostgresql(z *zalando.Postgresql) *zalando.Postgresql {
 	if p.HasSourceRanges() {
 		z.Spec.AllowedSourceRanges = p.Spec.AccessList.SourceRanges
 	}
 
-	z.Spec.NumberOfInstances = p.Spec.NumberOfInstances
-
-	// todo: Check if the validation should be performed here.
-	z.Spec.PostgresqlParam.PgVersion = p.Spec.Version
+	jsonZ, err := runtime.DefaultUnstructuredConverter.ToUnstructured(z)
+	if err != nil {
+		return nil, fmt.Errorf("failed to convert to unstructured zalando postgresql: %w", err)
+	}
+	jsonSpec, _ := jsonZ["spec"].(map[string]interface{})
+	// todo: Fix it in another branch.
+	// jsonSpec["maintenanceWindows"] = p.Spec.Maintenance
+	delete(jsonSpec, "maintenanceWindows")
 
-	z.Spec.ResourceRequests.CPU = p.Spec.Size.CPU
-	z.Spec.ResourceRequests.Memory = p.Spec.Size.Memory
-	z.Spec.ResourceLimits.CPU = p.Spec.Size.CPU
-	z.Spec.ResourceLimits.Memory = p.Spec.Size.Memory
+	// Delete unused fields
+	delete(jsonSpec, "clone")
```

Comment:
better check before deletion, otherwise we could be in for some surprises in the future :-)